fili-core/src/test/groovy/com/yahoo/bard/webservice/web/endpoints/LookupDimensionFilteringDataServletSpec.groovy

```diff
@@ -0,0 +1,287 @@
+package com.yahoo.bard.webservice.web.endpoints
+
+import com.yahoo.bard.webservice.data.dimension.BardDimensionField
+import com.yahoo.bard.webservice.data.dimension.DimensionDictionary
+import com.yahoo.bard.webservice.util.GroovyTestUtils
+import com.yahoo.bard.webservice.util.JsonSortStrategy
+
+/**
+ * Testing Query Time Lookup with extraction filter on Lookup Dimensions.
+ */
+class LookupDimensionFilteringDataServletSpec extends BaseDataServletComponentSpec{
+
+    @Override
+    def setup() {
+        DimensionDictionary dimensionStore = jtb.configurationLoader.dimensionDictionary
+        dimensionStore.findByApiName("sex").with {
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "sex1", "sex1Desc"))
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "sex2", "sex2Desc"))
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "sex3", "sex3Desc"))
+        }
+        dimensionStore.findByApiName("breed").with {
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "breed1", "breed1Desc"))
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "breed2", "breed2Desc"))
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "breed3", "breed3Desc"))
+        }
+        dimensionStore.findByApiName("species").with {
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "species1", "species1Desc"))
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "species2", "species2Desc"))
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "species3", "species3Desc"))
+        }
+    }
+
+    @Override
+    boolean compareResult(String result, String expectedResult, JsonSortStrategy sortStrategy) {
+        // Using default SORT_MAPS provided by Util
+        GroovyTestUtils.compareJson(result, expectedResult)
+    }
+
+    @Override
+    Class<?>[] getResourceClasses() {
+        [DataServlet.class]
+    }
+
+    @Override
+    String getTarget() {
+        return "data/pets/day/sex/breed/species"
```

Comment:
Since we're just focused on the filtering in this test, we shouldn't do grouping as well. As written, this test is actually doing _both_ the grouping _and_ the filtering using the namespaced dimensions. I _like_ the test, but either it's name should change to reflect that it's doing both, or we should remove the grouping.